tests/listediting.js

```diff
@@ -365,6 +365,160 @@ describe( 'ListEditing', () => {
 
 				expect( getModelData( model, { withoutSelection: true } ) ).to.equal( expectedModelData );
 			} );
+
+			describe( 'block elements inside list items', () => {
+				describe( 'single block', () => {
+					test(
+						'single item',
+						'<ul><li><p>Foo</p></li></ul>',
+						'<p>Foo</p>'
```

Comment:
I don't think this is a desirable result :(.

Reply:
We have concluded with @scofalik and @Reinmar that this indeed is not a result we would like to see for such cases, but the purpose of this PR was preserving order of block elements inside list items during conversion not improving conversion itself - this was extracted to a follow-up issue https://github.com/ckeditor/ckeditor5/issues/1318.